datalad/distribution/get.py

```diff
@@ -720,11 +728,19 @@ def __call__(
                             dsrec = content_by_ds.get(res['path'], set())
                             dsrec.update(res['contains'])
                             content_by_ds[res['path']] = dsrec
-                        if res.get('status', None) != 'notneeded':
-                            # all those messages on not having installed anything
-                            # are a bit pointless
-                            # "notneeded" for annex get comes below
+
+                        # we want to yield dataset results only once (not on every input path) and
+                        # in addition "notneeded" results are relevant only, if those paths were
+                        # actually requested:
+                        if res['path'] not in yielded_dataset_paths and \
+                                (res.get('status', None) != 'notneeded' or res.get('path', None) in input_paths):
                             yield res
```

Comment:
Since I have little knowledge of possible 'status' values, not sure if my comment has value -- if there is more than a single other than 'notneeded' value is possible here -- wouldn't we then yield only one (first) `res` and just swallow another with possibly different `"status"`? does it matter? if does -- then `yielded_dataset_paths` should memoize not only paths but `(path, status)` pairs.

Reply:
Not sure, whether I really get what you mean. We would yield only the first dataset result for a particular path - yes. I don't think it really matters. However, theoretically a dataset we just installed successfully could be possibly be screwed in a way that we neither would discover it to be "notneeded" nor be able to install again (yielding "ok"), but fail with "impossible" or "error". All that within a single run of that command.
Possible - so kind of agree. But also: Not really testable. Just saying. We can't produce such a situation without destroying the dataset(s) we are operating on in a parallel job. Even if we had that, timing would be relevant.